main.go

```diff
@@ -21,31 +21,36 @@ var release = "v1.1.0"
 var buildVersion = release + "-" + commit
 
 func main() {
+	e := make(chan *echo.Echo, 1)
+	q := make(chan int, 1)
+	os.Exit(realMain(e, q))
+}
+
+func realMain(echoChan chan<- *echo.Echo, quitChan chan int) int {
 	logger := log.New()
 
 	viper.SetEnvPrefix("logproxy")
 	viper.SetDefault("syslog", true)
 	viper.SetDefault("ironio", false)
 	viper.AutomaticEnv()
 
-	enableIronIO := viper.GetBool("syslog")
-	enableSyslog := viper.GetBool("ironio")
+	enableIronIO := viper.GetBool("ironio")
+	enableSyslog := viper.GetBool("syslog")
 
 	logger.Infof("logproxy %s booting", buildVersion)
 	if !enableIronIO && !enableSyslog {
-		logger.Fatalf("both syslog and ironio drains are disabled")
+		logger.Errorf("both syslog and ironio drains are disabled")
+		quitChan <- 1
+		return 1
 	}
 
-	// PHLogger
-	phLogger, err := setupPHLogger(http.DefaultClient, logger, buildVersion)
-	if err != nil {
-		logger.Fatalf("Failed to setup PHLogger: %s", err)
-	}
+	var messageQueue handlers.Queue
 
 	// RabbitMQ
-	consumer, producer, err := setupConsumerProducer(phLogger, buildVersion)
+	messageQueue, err := queue.NewRabbitMQQueue()
 	if err != nil {
-		logger.Fatalf("Failed to create consumer/producer: %v", err)
+		messageQueue, _ = queue.NewChannelQueue()
+		logger.Info("using internal channel queue")
```

Comment:
Should not default to internal channel queue. if Queue not found and queue expected then it should stop working so that issue can be remedied by devops.

Reply:
will make Go channel queue opt-in